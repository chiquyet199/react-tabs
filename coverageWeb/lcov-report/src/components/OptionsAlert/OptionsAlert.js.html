<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/OptionsAlert/OptionsAlert.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/OptionsAlert</a> OptionsAlert.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styled from "styled-components/native";
import {themed} from "../../utils/common";
import {Button} from "../Button";
&nbsp;
export const OptionsAlertContainer = styled.View`
  flex: 1;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.4);
  z-index: 10;
`;
&nbsp;
export const OptionsAlertContent = themed(styled.View`
  z-index: 11;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
  max-height: 168px;
  position: absolute;
  bottom: 0;
  height: 100%;
  margin-bottom: 20px;
  background-color: ${<span class="fstat-no" title="function not covered" >({</span>theme, globaltheme}) =&gt;
<span class="cstat-no" title="statement not covered" >    theme[globaltheme].altTextColor}</span>;
`);
&nbsp;
export const StyledButton = styled(Button)`
  height: 80px;
`;
&nbsp;
export const StyledView = styled.View`
  height: 80px;
`;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Mar 28 2019 14:23:06 GMT+0700 (Indochina Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
