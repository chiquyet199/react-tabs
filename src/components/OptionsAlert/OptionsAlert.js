import styled from "styled-components/native";
import {themed} from "../../utils/common";
import {Button} from "../Button";

export const OptionsAlertContainer = styled.View`
  flex: 1;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.4);
  z-index: 10;
`;

export const OptionsAlertContent = themed(styled.View`
  z-index: 11;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
  max-height: 168px;
  position: absolute;
  bottom: 0;
  height: 100%;
  margin-bottom: 20px;
  background-color: ${({theme, globaltheme}) =>
    theme[globaltheme].altTextColor};
`);

export const StyledButton = styled(Button)`
  height: 80px;
`;

export const StyledView = styled.View`
  height: 80px;
`;
