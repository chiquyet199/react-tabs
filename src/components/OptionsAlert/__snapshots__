/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/OptionsAlert/__native_snapshots__