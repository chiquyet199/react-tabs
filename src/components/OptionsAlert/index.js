import React from "react";
import {themed} from "../../utils/common";
import {Modal} from "react-native";
import {
  OptionsAlertContainer,
  OptionsAlertContent,
  StyledButton,
} from "./OptionsAlert";
import {bool, func, string, arrayOf, shape, oneOf} from "prop-types";

export const OptionsAlert = ({isOpen, onRequestClose, data}) => {
  const renderButtons = values => {
    return values.map(value => (
      <StyledButton
        key={value.title}
        type={value.type}
        middle={value.title}
        onClick={value.path}
      />
    ));
  };

  return (
    isOpen && (
      <Modal transparent onRequestClose={onRequestClose}>
        <OptionsAlertContainer onPress={onRequestClose}>
          <OptionsAlertContent>{renderButtons(data)}</OptionsAlertContent>
        </OptionsAlertContainer>
      </Modal>
    )
  );
};

OptionsAlert.propTypes = {
  isOpen: bool.isRequired,
  onRequestClose: func.isRequired,
  globaltheme: oneOf(["dark", "light"]).isRequired,
  data: arrayOf(
    shape({
      title: string.isRequired,
      path: func.isRequired,
      type: string.isRequired,
    }).isRequired,
  ),
};

export default themed(OptionsAlert);
