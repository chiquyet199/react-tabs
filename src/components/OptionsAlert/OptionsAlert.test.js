import React from "react";
import {OptionsAlert} from "./index";
import {Modal} from "react-native";
import {shallow} from "enzyme";
import {extend} from "underscore";
import {StyledButton} from "./OptionsAlert";

describe("Options Alert component", () => {
  test("Options Alert snapshot test", () => {
    const component = getComponent({isOpen: true});
    expect(component).toMatchSnapshot();
  });

  it("renders `OptionsAlert` when `isOpen` is true", () => {
    const component = getComponent({isOpen: true});
    expect(component.find(Modal)).toExist();
  });

  it("doesn't render `OptionsAlert` when `isOpen` is false", () => {
    const component = getComponent({isOpen: false});
    expect(component.find(Modal)).not.toExist();
  });

  it("passes correct props from data object to confirmation button", () => {
    const component = getComponent({
      isOpen: true,
    });
    const confirmationButton = component
      .find(StyledButton)
      .findWhere(button => button.prop("type") === "confirmation_primary");
    expect(confirmationButton).toHaveProp("onClick", mockPassportFunction);
    expect(confirmationButton).toHaveProp("middle", "Passport");
  });

  it("passes correct props from data object to cancel button", () => {
    const component = getComponent({
      isOpen: true,
    });
    const cancelButton = component
      .find(StyledButton)
      .findWhere(button => button.prop("type") === "confirmation_cancel");
    expect(cancelButton).toHaveProp("onClick", mockCancelFunction);
    expect(cancelButton).toHaveProp("middle", "Cancel");
  });

  const mockProps = () => ({
    isOpen: false,
    onRequestClose: () => {},
    globaltheme: "light",
    data: [
      {
        title: "Passport",
        path: mockPassportFunction,
        type: "confirmation_primary",
      },
      {
        title: "Cancel",
        path: mockCancelFunction,
        type: "confirmation_cancel",
      },
    ],
    theme: {},
  });

  const mockPassportFunction = jest.fn();

  const mockCancelFunction = jest.fn();

  const getComponent = props => {
    const parsedProps = extend(mockProps(), props);

    return shallow(<OptionsAlert {...parsedProps} />);
  };
});
