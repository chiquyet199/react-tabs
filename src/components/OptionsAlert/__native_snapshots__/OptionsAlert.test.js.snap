// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Alert component Options Alert snapshot test 1`] = `
<Component
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <Styled(Component)
    onPress={[Function]}
  >
    <Connect(WithTheme(Styled(Component)))>
      <Styled(Button)
        key="Passport"
        middle="Passport"
        onClick={[MockFunction]}
        type="confirmation_primary"
      />
      <Styled(Button)
        key="Cancel"
        middle="Cancel"
        onClick={[MockFunction]}
        type="confirmation_cancel"
      />
    </Connect(WithTheme(Styled(Component)))>
  </Styled(Component)>
</Component>
`;
